### PR TITLE
Add Work Flow Dispatch To Soaking Tests

### PR DESCRIPTION
**Pull request by sethAmazon**
**Description:** 
Add workflow dispatch to soaking test workflow

### CODE REVIEW

**Comment by jefchien** [comment]
Could we add an input to the `workflow_dispatch` to allow us to select a specific commit hash to run the soaking test against?

File: .github/workflows/soaking.yml [hunk 1]
```diff
@@ -58,7 +63,8 @@ jobs:
       commit_id: ${{ steps.get-testing-version.outputs.commit_id }}
     steps:
       - uses: actions/checkout@v2
-      
+        with:
+          fetch-depth: 0
```
Comment (jefchien) [inline .github/workflows/soaking.yml:67]: Why are we changing this?

File: .github/workflows/soaking.yml [hunk 2]
```diff
@@ -20,6 +20,11 @@ on:
   # or we can manually trigger this workflow by using dispatch for debuging
   repository_dispatch:
     types: [bump-version]
+  workflow_dispatch:
+    inputs:
+      sha:
+        description: 'Commit Hash You Want To Soak Must Be The Full SHA'
```
Comment (jefchien) [inline .github/workflows/soaking.yml:26]: nitpick: For consistency with other workflows. Feel free to ignore.

https://github.com/aws-observability/aws-otel-collector/blob/f15d6496e42745da36bb657f755601cd43b63856/.github/workflows/CD.yml#L14-L24

```suggestion
        description: 'the commit hash to soak test'
```

File: .github/workflows/soaking.yml (hunk 1 above)
Comment (sethAmazon) [reply .github/workflows/soaking.yml:67]: Not needed will remove

**Review by jefchien — APPROVED** [review]

**Pull request merged**